### PR TITLE
Implement color option

### PR DESCRIPTION
**Pull request by cmeeren**
Use of colors should be optional, and grey should be the default, as mentioned in https://github.com/Microsoft/visualfsharp/pull/3608#issuecomment-331988205.

Progress:

- [x] Implement color option
- [x] Change default color to what VS uses for its normal CodeLens (#999999 / 153, 153, 153)
- [x] Make sure the hyperlinks also follow the color above.

### CODE REVIEW

**Comment by cmeeren** [comment]
I think I have fixed the colors. The grey color is now the same as the default CodeLens.

With color:

image


image


Without color:

image

image

**Comment by cmeeren** [comment]
To be clear, as far as I'm concerned this is ready to be merged.

File: vsintegration/src/FSharp.Editor/CodeLens/LineLens.fs [hunk 1]
```diff
@@ -350,10 +350,29 @@ type internal FSharpCodeLensService
...
-            textBox.Inlines.Add (Documents.Run Settings.CodeLens.Prefix)
+
+            let prefix = Documents.Run Settings.CodeLens.Prefix
+            prefix.Foreground <- SolidColorBrush(Color.FromRgb(153uy, 153uy, 153uy))
+            textBox.Inlines.Add (prefix)
+
             for text in taggedText do
+
+                let properties = layoutTagToFormatting text.Tag
+                let properties =
```
Comment (realvictorprm) [inline vsintegration/src/FSharp.Editor/CodeLens/LineLens.fs]: merge those two definitions, it's not needed to this here.

File: vsintegration/src/FSharp.Editor/CodeLens/LineLens.fs [hunk 2]
```diff
@@ -350,10 +350,29 @@ type internal FSharpCodeLensService
...
+                    if Settings.CodeLens.UseColors
+                    then
+                        match properties.ForegroundBrush with
+                        | :? SolidColorBrush as b ->
+                            let c = b.Color
+                            // Change to correct gray color
+                            if c.R = c.G && c.R = c.B then properties.SetForeground(Color.FromRgb(153uy, 153uy, 153uy)) else properties
```
Comment (realvictorprm) [inline vsintegration/src/FSharp.Editor/CodeLens/LineLens.fs]: We should make this customizable.

File: vsintegration/src/FSharp.Editor/CodeLens/LineLens.fs [hunk 3]
```diff
@@ -362,10 +381,11 @@ type internal FSharpCodeLensService
                             navigation.NavigateTo nav.Range)
                         h :> Documents.Inline
                     | _ -> run :> _
+                DependencyObjectExtensions.SetTextProperties (inl, properties)
                 textBox.Inlines.Add inl
             lens.Computed <- true
             lens.UiElement <- textBox
-        }
```
Comment (realvictorprm) [inline vsintegration/src/FSharp.Editor/CodeLens/LineLens.fs]: remove this change please

**Review by realvictorprm — CHANGES_REQUESTED** [review]
Small suggestions.

File: vsintegration/src/FSharp.Editor/CodeLens/LineLens.fs (hunk 1 above)
Comment (cmeeren) [reply vsintegration/src/FSharp.Editor/CodeLens/LineLens.fs]: How would you want them merged? The first definition of `properties` is used several places in the second definition. Would a name change to the first one suffice?

File: vsintegration/src/FSharp.Editor/CodeLens/LineLens.fs (hunk 2 above)
Comment (cmeeren) [reply vsintegration/src/FSharp.Editor/CodeLens/LineLens.fs]: Sure, but how? Is there a color picker control we can use? (I've no experience with WPF.)

**Comment by cmeeren** [comment]
Sorry for all the commits, got bitten by Git while trying to clean up...

File: vsintegration/src/FSharp.Editor/CodeLens/LineLens.fs (hunk 1 above)
Comment (cmeeren) [reply vsintegration/src/FSharp.Editor/CodeLens/LineLens.fs]: Fixed :)

**Review by realvictorprm — APPROVED** [review]
Very well, thanks!

**Pull request merged**